File: Lecture2/data_visualization/script/import_data.R
# This script demonstrates various methods to import data in R

# Section 1: Set Working Directory
setwd("/home/hoangnguyen/Documents/data_visualization")  # Set working directory
getwd()  # Check working directory

# Section 2: Import .txt Files
data_1 <- read.table("data/other_data/text.txt", sep = "\t", header = T, check.names = F)  # Import .txt file
data_2 <- read.table("data/other_data/smoking.txt", sep ="\t", header = T, check.names = F)  # Import another .txt file
head(data_2)  # Display first few rows of the data

# Section 3: Import .CSV Files
data_3 <- read.csv("data/other_data/prawnGR.CSV", sep=",")  # Import .CSV file
head(data_3)  # Display first few rows of the data
dt <- read.table("data/other_data/prawnGR.CSV", sep=",")  # Import .CSV file using read.table()
dt  # Display the data

# Section 4: Import .tsv Files
data_4 <- read.delim("data/other_data/Petal.Length.tsv", sep="\t")  # Import .tsv file
head(data_4)  # Display first few rows of the data

# Section 5: Import .xls Files
install.packages("xlsx")  # Install the 'xlsx' package
library(xlsx)  # Load the 'xlsx' package
data_4 <- read_excel("data/other_data/whaledata.xls", sheet = 1)  # Import .xls file

# Section 6: Using data.table Package
library(data.table)  # Load the 'data.table' package
data_5 <- as.data.frame(fread("data/other_data/Petal.Length.tsv"))  # Import .tsv file using fread()
data_5  # Display the data
data_6 <- as.data.frame(fread("data/other_data/children.txt"))  # Import .txt file using fread()
data_6  # Display the data
data_7 <- as.data.frame(fread("data/other_data/Gigartina.CSV"))  # Import .CSV file using fread()
data_7  # Display the data

# Section 7: Read Multiple Files in R Using apply and reduce
list_file <- list.files("data/multi_files", pattern = ".tsv", full.names = T)  # List all .tsv files in the directory

read_file <- lapply(list_file, function(x) read.delim(x, header = T, sep = "\t"))  # Read all files using lapply()

data <- Reduce(function(x, y) merge(x, y, by = "ID"), read_file)  # Merge all files using Reduce()

data_rm <- data[!duplicated(as.list(data))]  # Remove duplicate rows

# Section 8: Save the Data
write.table(data_rm, paste0("output", "/newdata.tsv"), col.names = T, row.names = F, sep = "\t", quote = F)  # Save the data to a .tsv file

File: Lecture2/data_visualization/script/import_data.py
import pandas as pd  # Import pandas library

# Section 1: Set Working Directory
# In Python, you can change the working directory using os module
import os
os.chdir("/home/hoangnguyen/Documents/data_visualization")  # Set working directory
os.getcwd()  # Check working directory

# Section 2: Import .txt Files
data_1 = pd.read_csv("data/other_data/text.txt", sep="\t")  # Import .txt file
data_2 = pd.read_csv("data/other_data/smoking.txt", sep="\t")  # Import another .txt file
print(data_2.head())  # Display first few rows of the data

# Section 3: Import .CSV Files
data_3 = pd.read_csv("data/other_data/prawnGR.CSV")  # Import .CSV file
print(data_3.head())  # Display first few rows of the data

# Section 4: Import .tsv Files
data_4 = pd.read_csv("data/other_data/Petal.Length.tsv", sep="\t")  # Import .tsv file
print(data_4.head())  # Display first few rows of the data

# Section 5: Import .xls Files
data_4 = pd.read_excel("data/other_data/whaledata.xls", sheet_name = 0)  # Import .xls file

# Section 6: Python doesn't have an equivalent to R's fread function for fast file reading.
# But pandas provides a similar function 'read_csv' for reading .csv files.

data_5 = pd.read_csv("data/other_data/Petal.Length.tsv", sep="\t")  # Import .tsv file using read_csv()
print(data_5)  # Display the data

data_6 = pd.read_csv("data/other_data/children.txt", sep="\t")  # Import .txt file using read_csv()
print(data_6)  # Display the data

data_7 = pd.read_csv("data/other_data/Gigartina.CSV")  # Import .CSV file using read_csv()
print(data_7)  # Display the data

# Section 7: Read Multiple Files in Python Using list comprehension
list_file = [f for f in os.listdir("data/multi_files") if f.endswith('.tsv')]  # List all .tsv files in the directory

read_file = [pd.read_csv(file, sep="\t") for file in list_file]  # Read all files using list comprehension

data = pd.concat(read_file, axis=0, ignore_index=True)  # Merge all files using concat()

data_rm = data.drop_duplicates()  # Remove duplicate rows

# Section 8: Save the Data
data_rm.to_csv("output/newdata.tsv", index=False, sep="\t")  # Save the data to a .tsv file
